webapp/channels/src/components/sidebar/sidebar_category/sidebar_category_menu/mark_as_read_menu_item.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import * as Menu from 'components/menu';
+
+import {MarkAsUnreadIcon} from '@mattermost/compass-icons/components';
+import {FormattedMessage} from 'react-intl';
+import {useDispatch} from 'react-redux';
+import {openModal} from 'actions/views/modals';
+import {ModalIdentifiers} from 'utils/constants';
+import MarkAsReadConfirmModal from './mark_as_read_confirm_modal';
+
+type Props = ({
+    id: string;
+    handleViewCategory: () => void;
+    numChannels: number;
+})
+
+const MarkAsUnreadItem = ({
+    id,
+    handleViewCategory,
+    numChannels,
+}: Props) => {
+    const dispatch = useDispatch();
+
+    const onClick = useCallback(() => {
+        if (numChannels <= 0) {
+            return;
+        }
+
+        if (numChannels === 1) {
+            handleViewCategory();
+            return;
+        }
+
+        dispatch(openModal({
+            modalId: ModalIdentifiers.DELETE_CATEGORY,
+            dialogType: MarkAsReadConfirmModal,
+            dialogProps: {
+                handleConfirm: handleViewCategory,
+                numChannels,
+            },
+        }));
+    }, [dispatch, handleViewCategory, numChannels]);
+
+    return (
+        <Menu.Item
+            id={`view-${id}`}
+            onClick={onClick}
+            aria-haspopup={true}
```

Comment:
Now it has a popup, since now it shows the confirmation modal, right?

But since it doesn't show always, I guess I would have to change the aria depending on the number of channels to mark as read.